src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,136 @@
+            } else if (time.matches("([0-9]{4})-([0-9]{2})-([0-9]{2})")) {
+                return LocalDate.parse(time, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            } else if (time.matches("([0-9]{4}) ([0-9]{2}) ([0-9]{2})")) {
+                return LocalDate.parse(time, DateTimeFormatter.ofPattern("yyyy MM dd"));
+            } else if (time.matches("([0-9]{2}) ([0-9]{2}) ([0-9]{4})")) {
+                return LocalDate.parse(time, DateTimeFormatter.ofPattern("dd MM yyyy"));
+            } else {
+                throw new DukeException("☹ OOPS!!! This is not a proper time format, "
+                        + "please refer to command format information.");
+            }
+        } catch (Exception e) {
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+    /**
+     * Returns simplified command string parsed from a user command.
+     * If the command is invalid, exception is thrown.
+     *
+     * @param line a string containing a user command.
+     * @param taskList a List of tasks.
+     * @return command string understandable by other methods/classes.
+     * @throws DukeException If the command is regarded invalid/incomplete by duke chatbot.
+     */
+    public String parseCommand(String line, TaskList taskList) throws DukeException {
+        String[] taskString = line.split(" ", 2);
+        String taskType = taskString[0];
+        if (taskString.length == 1 && !taskType.equals("list") && !taskType.equals("bye") && !taskType.equals("help")) {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        } else if (taskType.equals("todo")) {
+            String description = taskString[1].trim();
+            if (description.equals("")) { // task description empty
+                throw new DukeException("☹ OOPS!!! The description of a Todo task cannot be empty.");
+            }
+            taskList.addTask(new ToDo(description));
+            return String.format("todo %d", taskList.getSize());
+        } else if (taskType.equals("deadline")) {
+            if (!taskString[1].contains("/by")) { // command not properly formatted
+                throw new DukeException("☹ OOPS!!! The command not properly formatted.\n"
+                        + "Please follow the format: Deadline {description} /by {date}.");
+            }
+            String[] descriptions = taskString[1].split("/by");
+            String description = descriptions[0].trim();
+            LocalDate end = parseDate(descriptions[1]);
+            if (description.equals("")) { // task description empty
+                throw new DukeException("☹ OOPS!!! The description of a Deadline task cannot be empty.");
+            }
+            taskList.addTask(new Deadline(description, end));
+            return String.format("deadline %d", taskList.getSize());
+        } else if (taskType.equals("event")) {
```

Comment:
Seems like after each "if" or "else if" statement, there seem to be many lines of code manipulating the taskString. Perhaps can consider placing all these code into a method in the parser class(after the conditional statement), so that the code here would be more readable